database/scripts/main.js

```diff
@@ -126,7 +126,18 @@ function createPostElement(postId, title, text, author) {
   // Set values.
   postElement.getElementsByClassName('text')[0].innerText = text;
   postElement.getElementsByClassName('mdl-card__title-text')[0].innerText = title;
-  postElement.getElementsByClassName('username')[0].innerText = author;
+  postElement.getElementsByClassName('username')[0].innerText = author || 'Anonymous';
+
+  // Retrieve the post's user id
+  firebase.database().ref('posts/' + postId + '/uid').once('value').then(function(snapshot) {
```

Comment:
In this example we're trying to avoid having to fetch the user information by using fanned out data: basically we save (duplicate) minimal information that we need about the author in the post data. (kind of the 'author' filed which is the author name. 
You need to add an extra field which is `authorPic` (for instance) in the function `writeNewPost` and use that.